tests/testmongodb.nim

```diff
@@ -0,0 +1,219 @@
+import unittest
+
+import os, strutils, sequtils, times
+
+import norm / mongodb
+
+
+const
+  # for local testing, modify your /etc/hosts file to contain "mongodb_1"
+  # pointing to your local mongodb server
+  dbConnection = "mongodb://mongodb_1:27017"
+  dbName = "TestDb"
+  customDbName = "TestCustomDb"
+
+type
+  Copyright = object
+    legal_name: string
+    year: int
+    # year {.dbCol: "yr".}: int
+
+dbAddObject(Copyright)
+
+db(dbConnection, "", "", dbName):
+  type
+    User {.table: "users".} = object
+      email {.unique.}: string
+      ssn: Option[int]
+      birthDate: Time
+    Publisher {.table: "publishers".} = object
+      title {.unique.}: string
+    Book {.table: "books".} = object
+      title: string
+      authorEmail {.fk: User.email, onDelete: "CASCADE".}: string
+      publisherTitle {.fk: Publisher.title.}: string
+      ratings: seq[float]
+      copyright: Copyright
+
+  # TODO proc getBookById(id: Oid): Book = withDb(Book.getOne id)
+  # TODO add and test foreign key pulls in mongo
+
+type
+  Edition {.table: "editions".} = object
+    id {.dbCol: "_id".}: Oid
+    title: string
+    book: Book
+
+dbAddCollection(Edition)
+
+suite "Creating and dropping tables, CRUD":
+  setup:
+    var 
+      user_reference_id: array[1..9, Oid]
+      publisher_reference_id: array[1..9, Oid]
+      book_reference_id: array[1..9, Oid]
+      edition_reference_id: array[1..9, Oid]
+
+    withDb:
+      createTables(force=true)

```

Comment:
I wonder if this will cause confusion... Mongo doesn't have “tables,” it has “collections.” Maybe, introduce an alias `createCollections`?

Reply:
I will do that.